### ISSUE TITLE
wrong `destinationIndex` calculation when dropping to the root

### ISSUE THREAD

**Issue by 1998MaxChykalo** [minop1205/react-dnd-treeview#129]
if I drag an element and drop it at the root level then `destinationIndex` is calculated as if every element is the direct child of the root element.
Example: 
<img width="600" alt="image" src="https://user-images.githubusercontent.com/27495063/184613298-5eb6d144-0f57-4386-bcd1-02d2e639418c.png">
the `destinationIndex` is 6. But we have only 5 elements in the root, so the `destinationIndex` in this case should've been 4: 
<img width="600" alt="image" src="https://user-images.githubusercontent.com/27495063/184613470-e8da5ec7-0cfb-4a37-a769-a8432aa3e060.png">
If we drop an element to some nested item, then `destinationIndex` is calculated relatively to the item it was dropped on.
link to sandbox: https://codesandbox.io/s/minimum-configuration-js-forked-n3yor6?file=/src/App.jsx

Expected result:
`destinationIndex` is calculated based on the level of nesting where it was dropped

**Comment by Surjit4048** [comment]
Any update on this? I am facing same issue.

**Comment by Surjit4048** [comment]
nvm, I can access parent and previous node with setup easy. All good for me.

**Comment by hightgha** [comment]
@minop1205 any update?
same issue

**Comment by a-utkarsh** [comment]
Is there any update or any workaround for this, I need to know the index of an item where it inserted with respect to its parent in case of sortable false. @minop1205 

Please update ASAP.

**Comment by a-utkarsh** [comment]
is there anyway to add id to the li element? @minop1205

**Comment by minop1205** [comment]
Sorry everyone for misleading you with the README explanation.
First, let us discuss destinationIndex.

If the `sort` property of the tree is `false`, the order of the nodes follows the order of the data array.Therefore, reordering the nodes requires reordering the data array.

`destinationIndex` is not an index value relative to the parent node, but rather an index value indicating where in the data array the drag source data is to be moved.

In fact, the array swapping is done inside the library, and the `onDrop` callback receives the reordered data array.

There is currently no way to get the index value relative to the parent node.
I will make it available in the next update.

Also, I'll correct the description of destinationIndex in the README.

If you don't mind me asking as a reference, what is the use of relative indexes?
For example, is this a case of sending a relative index to synchronize the tree sequence with the server?

**Comment by hightgha** [comment]
> Sorry everyone for misleading you with the README explanation. First, let us discuss destinationIndex.
> 
> If the `sort` property of the tree is `false`, the order of the nodes follows the order of the data array.Therefore, reordering the nodes requires reordering the data array.
> 
> `destinationIndex` is not an index value relative to the parent node, but rather an index value indicating where in the data array the drag source data is to be moved.
> 
> In fact, the array swapping is done inside the library, and the `onDrop` callback receives the reordered data array.
> 
> There is currently no way to get the index value relative to the parent node. I will make it available in the next update.
> 
> Also, I'll correct the description of destinationIndex in the README.
> 
> If you don't mind me asking as a reference, what is the use of relative indexes? For example, is this a case of sending a relative index to synchronize the tree sequence with the server?

@minop1205 in my case, i wanna to update order in database after reordering items.
Cuz destinationIndex is wrong, i'm calculating order in current lvl, by checking position in array. Ordering is applied in some lvl
For example: 
1 Parent1
   1 Child1
   2 Child2
2 Parent2
3 Parent3
   1 Child1
   2 Child2

**Comment by a-utkarsh** [comment]
@minop1205 Yes, I need to synchronize the tree sequence with the server.  Otherwise how will we know where the node is inserted when we are reordering within the same parent or different parent.
Anyway for those who are still stuck in this issue, I have found a workaround or hack you can say.
Here is the code for the same, You already have the drag node id and you can get the parent node of the same
Basically need to pass down the node and parent node to the list item and get all the items from the dom.
![image](https://user-images.githubusercontent.com/29899695/199299059-0918eb99-d464-462e-820f-3cf24fc33ade.png)
![image](https://user-images.githubusercontent.com/29899695/199299446-754c9575-22e1-4e29-9fb3-ae07526c03de.png)

**Comment by minop1205** [comment]
@hightgha @a-utkarsh Thank you for your comment!

I have just released v3.3.
In this version, an option called relativeIndex is available in addition to destinationIndex.

The relativeIndex indicates the index relative to the parent node.
The relativeIndex is actually equivalent to the index of the placeholder.

Below is an image of how it works.

![2022-11-02_04h53_42](https://user-images.githubusercontent.com/3772820/199330836-bb353b2c-61d9-487f-855c-1581431d5116.png)

I will Close this Issue, but reopen it if you have any problems.
